server/marketplace/plugins.go

```diff
@@ -0,0 +1,18 @@
+package marketplace
+
+import (
+	"github.com/mattermost/mattermost-server/model"
+)
+
+// Plugins is a list of Marketplace plugins.
+type Plugins []*model.BaseMarketplacePlugin
+
+// GetPlugin gets a plugin by id.
+func (p *Plugins) GetPlugin(id string) (plugin *model.BaseMarketplacePlugin, found bool) {
+	for _, plugin := range *p {
+		if plugin.Manifest.Id == id {
+			return plugin, true
+		}
+	}
+	return nil, false
```

Comment:
I think, returning `NotFound ` error makes the most sense since _marketplace-addon_ plugin might try to update some manually installed plugins that might not present in the Mattermost Marketplace.

As we don't have any info in the _model.Manifest_ about if a plugin installed from the Marketplace or not in the first place, having to return a dedicated error about this seems more explanatory.

I'll be doing the update by changing _found_ to _NotFound_ error.

Reply:
done by f42ce7592c68359e248e8db843cadd6cbe2bf34a

Reply:
I second read your root message about `InstallMarketplacePlugin()`. I'll be working on this again and maybe replace `NotFound` error with something else. Not sure yet.